pydantic/config.py

```diff
@@ -944,6 +944,21 @@ class Model(BaseModel):
         can be different depending on the Python version used.
     '''
 
+    cache_strings: bool | Literal['all', 'keys', 'none']
+    """
+    Whether to cache strings to avoid constructing new Python objects. Defaults to True.
+
+    Enabling this setting should significantly improve validation performance while increasing memory usage slightly.
```

Comment:
again add something about "... if repeated strings are very rare, it might be worth switching it off or using `keys`.